velox/type/TimestampConversion.cpp

```diff
@@ -528,10 +535,21 @@ int32_t getMaxDayOfMonth(int32_t year, int32_t month) {
 }
 
 int64_t daysSinceEpochFromDate(int32_t year, int32_t month, int32_t day) {
+  int64_t daysSinceEpoch;
+  auto status = daysSinceEpochFromDate(year, month, day, daysSinceEpoch);
```

Comment:
@mbasmanova `int64_t daysSinceEpochFromDate` is preserved here and the unit tests still use this API. Should I remove it?

Reply:
@marin-ma Is daysSinceEpochFromDate used only in unit tests? If so, then it is better to remove.

Reply:
@mbasmanova The previous one is also in use by non-test code. e.g. in sparksql function call: https://github.com/facebookincubator/velox/blob/b11aeeb796782285be895053d35751a5fd00da20/velox/functions/sparksql/DateTimeFunctions.h#L315-L334

Reply:
@marin-ma I see. Looks like we have inconsistent APIs. Some provide 2 versions: one that returns Status and doesn't throw and another that throws. While other APIs provide only non-throwing version. Would it make sense to align all these APIs to provide only non-throwing behavior?